salt/utils/win_functions.py

```diff
@@ -121,9 +121,16 @@ def get_current_user():
     '''
     try:
         user_name = win32api.GetUserNameEx(win32api.NameSamCompatible)
-        if user_name[-1] == '$' and win32api.GetUserName() == 'SYSTEM':
+        if user_name[-1] == '$':
             # Make the system account easier to identify.
-            user_name = 'SYSTEM'
+            # Fetch sid so as to handle other language than english
+            test_user = win32api.GetUserName()
+            sid, d, t = win32security.LookupAccountName('', test_user)
+            real_sid = user = win32security.ConvertSidToStringSid(sid)
+            if test_user == 'SYSTEM':
+                user_name = 'SYSTEM'
+            elif real_sid == 'S-1-5-18':
+                user_name = 'SYSTEM'
     except pywintypes.error as exc:
```

Comment:
Agreed, I'm going to rewrite this, and try to use the already existing function as well.